src/mongocxx/client.cpp

```diff
@@ -55,6 +55,16 @@ void* client::implementation() const {
     return _impl->client_t;
 }
 
+void client::read_concern(class read_concern rc) {
+    libmongoc::client_set_read_concern(_impl->client_t, rc._impl->read_concern_t);
+}
+
+class read_concern client::read_concern() const {
+    auto rc = libmongoc::client_get_read_concern(_impl->client_t);
+    return (class read_concern){
```

Comment:
the syntax for class-qualifying a type to call it's constructor is a bit confusing here as a cursory glance would lead one to believe you are deploying the dreaded c-style cast. Would this compile if we just return the initializer list? If not can you leave a comment explaining what is actually going on?